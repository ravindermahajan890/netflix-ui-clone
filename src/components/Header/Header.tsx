import PersonOutlineIcon from '@mui/icons-material/PersonOutline';
import { Box, Menu, MenuItem, styled } from '@mui/material';
import { useState } from 'react';
import { useAuth } from '../../providers/Auth';
import Logo from '../Logo';

const CustomMenu = styled(Menu)(({ theme }) => ({
  '& .MuiMenu-paper': {
    border: `1px solid ${theme.palette.lightGrey.main}`,
    borderRadius: '4px',
    background: theme.palette.primary.main,
    padding: '2px 20px',
  },
}));

const CustomMenuItem = styled(MenuItem)(({ theme }) => ({
  borderColor: theme.palette.lightGrey.main,
  justifyContent: 'center',
}));

const Header = () => {
  const { logout } = useAuth();

  const [menuAnchorElement, setMenuAnchorElement] = useState<null | SVGSVGElement>(null);
  const openMenu = ({ currentTarget }: { currentTarget: SVGSVGElement }) => setMenuAnchorElement(currentTarget);
  const closeMenu = () => setMenuAnchorElement(null);

  return (
    <Box display="flex" justifyContent="space-between" alignItems="center" px="40px" py="16px" bgcolor="primary.main">
      <Logo />
      <PersonOutlineIcon fontSize="large" onClick={openMenu} />
      <CustomMenu anchorEl={menuAnchorElement} open={Boolean(menuAnchorElement)} onClose={closeMenu}>
        <CustomMenuItem onClick={closeMenu} divider>
          Profile
        </CustomMenuItem>
        <CustomMenuItem onClick={closeMenu} divider>
          Settings
        </CustomMenuItem>
        <CustomMenuItem onClick={logout}>Logout</CustomMenuItem>
      </CustomMenu>
    </Box>
  );
};

export default Header;
