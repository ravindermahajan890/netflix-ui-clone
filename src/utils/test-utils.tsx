// eslint-disable-next-line import/no-extraneous-dependencies
import { render as rtlRender } from '@testing-library/react';
import { PropsWithChildren, ReactElement } from 'react';
import { HashRouter } from 'react-router-dom';
import Providers from '../providers';

// eslint-disable-next-line import/prefer-default-export
export const render = (ui: ReactElement) => {
  const wrapper = ({ children }: PropsWithChildren) => (
    <Providers>
      <HashRouter>{children}</HashRouter>
    </Providers>
  );

  return rtlRender(ui, { wrapper });
};
