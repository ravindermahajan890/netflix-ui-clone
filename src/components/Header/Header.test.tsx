import { render } from '../../utils/test-utils';
import Header from '.';

describe('Header', () => {
  it('should render', () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });
});
