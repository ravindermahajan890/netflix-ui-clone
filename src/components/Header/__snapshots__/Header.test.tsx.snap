// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 40px;
  padding-right: 40px;
  padding-top: 16px;
  padding-bottom: 16px;
  background-color: #000;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: left;
  -ms-flex-pack: left;
  -webkit-justify-content: left;
  justify-content: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 12px;
}

.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: Inter;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-3 {
  width: 100%;
  height: 100%;
  text-align: center;
  object-fit: cover;
  color: transparent;
  text-indent: 10000px;
}

.emotion-4 {
  margin: 0;
  font-family: Inter;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  color: #FE0039;
}

.emotion-5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 2.1875rem;
}

<div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular emotion-2"
      >
        <img
          alt="TAMASHFLIX"
          class="MuiAvatar-img emotion-3"
          src="/tamashflix.svg"
        />
      </div>
      <h5
        class="MuiTypography-root MuiTypography-h5 emotion-4"
      >
        TAMASHFLIX
      </h5>
    </div>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge emotion-5"
      data-testid="PersonOutlineIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
